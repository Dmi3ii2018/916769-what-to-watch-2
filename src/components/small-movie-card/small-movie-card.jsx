import React from "react";
import PropTypes from "prop-types";
import {VideoPreview} from "../video-preview/video-preview";
import {Redirect, withRouter} from "react-router-dom";
import {connect} from "react-redux";

import {ActionCreator} from "../../reducer/root-reducer";

class SmallMovieCard extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      isVideoPreviewPlaying: false,
      isCardClicked: this.props.isClicked,
    };

    this._filmCardOverHandler = this._filmCardOverHandler.bind(this);
    this._filmCardOutHandler = this._filmCardOutHandler.bind(this);
  }

  _filmCardOverHandler() {
    // this.props.onFilmCardOver(id);
    this.setState({isVideoPreviewPlaying: true});
  }

  _filmCardOutHandler() {
    // this.props.onFilmCardOut();
    this.setState({isVideoPreviewPlaying: false});
  }

  render() {
    const {name, img, id, preview, setCardClicked} = this.props;
    const {isVideoPreviewPlaying, isCardClicked} = this.state;

    if (isCardClicked) {
      return <Redirect to={`/films/${id}`} />;
    }

    return <article onClick={() => {
      setCardClicked(id);
      this.setState({isCardClicked: true});
    }}
    className="small-movie-card catalog__movies-card"
    onMouseOver={() => this._filmCardOverHandler(id)}
    onMouseOut={this._filmCardOutHandler}>
      {isVideoPreviewPlaying
        ? <VideoPreview
          previewSrc={preview} />
        : <>
          <div className="small-movie-card__image">
            <img src={img} alt={name} width="280" height="175" />
          </div>
          <h3 className="small-movie-card__title">
            <a className="small-movie-card__link" href="movie-page.html">{name}</a>
          </h3>
        </>
      }
    </article>;
  }
}

SmallMovieCard.propTypes = {
  name: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  poster: PropTypes.string,
  preview: PropTypes.string.isRequired,
  choosenFilm: PropTypes.number.isRequired,
  onFilmCardOver: PropTypes.func,
  onFilmCardOut: PropTypes.func,
  setCardClicked: PropTypes.func.isRequired,
};


const mapStateToProps = (state) => ({
  choosenFilm: state.filterReducer.choosenFilmId,
});

const mapDispatchToProps = (dispatch) => ({
  setCardClicked: (id) => {
    dispatch(ActionCreator.chooseFilm(id));
  }
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(SmallMovieCard));
