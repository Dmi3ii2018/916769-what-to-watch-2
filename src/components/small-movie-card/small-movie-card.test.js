import React from "react";
import {shallow} from "enzyme";
// import renderer from "react-test-renderer";
import {SmallMovieCard} from "./small-movie-card";

it(`Small movie card renders correctly`, () => {
  const tree = shallow(<SmallMovieCard
    name = {`Fantastic Beasts: The Crimes of Grindelwald`}
    img = {`fantastic-beasts-the-crimes-of-grindelwald`}
    id = {1}
    onFilmCardOver = {jest.fn()}
    onFilmCardOut = {jest.fn()}
  />);

  expect(tree).toMatchSnapshot();
});
