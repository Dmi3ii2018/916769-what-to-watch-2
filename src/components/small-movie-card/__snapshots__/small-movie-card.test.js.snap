// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Small movie card renders correctly 1`] = `ShallowWrapper {}`;
